src/bootstrap/builder.rs

```diff
@@ -2309,6 +2324,11 @@ impl From<Cargo> for Command {
             cargo.command.env("RUSTDOCFLAGS", rustdocflags);
         }
 
+        let hostflags = cargo.hostflags;
+        for (index, flag) in hostflags.rustc.into_iter().enumerate() {
+            cargo.command.env(format!("RUSTC_HOST_CMD_{index}"), flag);
+        }
```

Comment:
Can we somehow send all the values over a single env variable without indexing? e.g., by sending values with a special character("ideally ',' or a whitespace") as a seperator like `ENV_VAL="-Clink-args=-fuse-ld=lld,-Ctarget-feature=-crt-static"` so on the shim side all we need to do is read the single env value and split them by that special character.

Reply:
If we do that we can avoid iterating over all the existing env variables here https://github.com/rust-lang/rust/blob/a3fd45df34bcc6c25f43bcd646879364a72f83d5/src/bootstrap/bin/rustc.rs#L117-L119

Reply:
Probably not a comma, since e.g. linker flags can look like `-Wl,q`. Maybe a space could be enough, although I'm not sure if a space in the arguments ever makes sense (?). I could use a space and add an assert that there's no space in the values passed from the bootstrap side. What do you think about that?

Reply:
Sounds good to me :)

Reply:
Hmm. How do you split an `OsString` by spaces :)

Reply:
Casting it into String should be fine